x-pack/plugins/fleet/server/services/epm/archive/validation.ts

```diff
@@ -202,40 +244,87 @@ export function parseAndVerifyDataStreams(
 
     const {
       title: dataStreamTitle,
-      release,
+      release = DEFAULT_RELEASE_VALUE,
       type,
       dataset,
-      ingest_pipeline: ingestPipeline,
       streams: manifestStreams,
+      elasticsearch,
       ...restOfProps
     } = manifest;
     if (!(dataStreamTitle && type)) {
       throw new PackageInvalidArchiveError(
         `Invalid manifest for data stream '${dataStreamPath}': one or more fields missing of 'title', 'type'`
       );
     }
+
+    let ingestPipeline;
+    const ingestPipelinePaths = paths.filter((path) =>
+      path.startsWith(`${pkgKey}/data_stream/${dataStreamPath}/elasticsearch/ingest_pipeline`)
+    );
+
+    if (
+      ingestPipelinePaths.length &&
+      (ingestPipelinePaths.some((ingestPipelinePath) =>
+        ingestPipelinePath.endsWith(DEFAULT_INGEST_PIPELINE_FILE_NAME_YML)
+      ) ||
+        ingestPipelinePaths.some((ingestPipelinePath) =>
+          ingestPipelinePath.endsWith(DEFAULT_INGEST_PIPELINE_FILE_NAME_JSON)
+        ))
+    ) {
+      ingestPipeline = DEFAULT_INGEST_PIPELINE_VALUE;
+    }
+
     const streams = parseAndVerifyStreams(manifestStreams, dataStreamPath);
 
+    const parsedElasticsearchEntry: Record<string, any> = {};
+
+    if (ingestPipeline) {
+      parsedElasticsearchEntry['ingest_pipeline.name'] = DEFAULT_INGEST_PIPELINE_VALUE;
+    }
+
+    if (elasticsearch?.privileges) {
+      parsedElasticsearchEntry.privileges = elasticsearch.privileges;
+    }
+
+    if (elasticsearch?.index_template?.mappings) {
+      parsedElasticsearchEntry['index_template.mappings'] = expandDottedEntries(
```

Comment:
These mappings/settings fields come through the manifest with `dotted.key.names`, but the package info object expects them to be expanded to nested object structures.